src/view/xwayland.cpp

```diff
@@ -115,15 +115,17 @@ class wayfire_xwayland_view_base : public wf::wlr_view_t
          * coordinates */
         if (get_output())
         {
+            auto current_workspace = get_output()->workspace->get_current_workspace();
+            auto wsize = get_output()->workspace->get_workspace_grid_size();
             auto og = get_output()->get_layout_geometry();
             configure_geometry.x -= og.x;
             configure_geometry.y -= og.y;
-
-            /* It is possible the client requests to position itself on another
-             * output. However, we want to make sure that the view stays on its
-             * output, since this is how Wayland works */
-            configure_geometry = wf::clamp(configure_geometry,
-                get_output()->get_relative_geometry());
+            wlr_box wsg{
```

Comment:
Can we have a comment explaining why we do this (we do it so that the client doesn't go off the screen)?